workflow/executor/pns/pns.go

```diff
@@ -418,18 +418,22 @@ func parseContainerIDFromCgroupLine(line string) string {
 		if containerID := parts[len(parts)-1]; containerID != "" {
 			// need to check for empty string because the line may look like: 5:rdma:/
 
-			// for crio we need to get rid of "crio-" prefix and ".scope" suffix
-			// e.g. crio-7a92a067289f6197148912be1c15f20f0330c7f3c541473d3b9c4043ca137b42.scope
-			containerID := strings.TrimSuffix(strings.TrimPrefix(containerID, "crio-"), ".scope")
+			// remove possible ".scope" suffix
+			containerID := strings.TrimSuffix(containerID, ".scope")
 
 			// for compatibility with cri-containerd record format when using systemd cgroup path
 			// example record in /proc/{pid}/cgroup:
 			// 9:cpuset:/kubepods-besteffort-pod30556cce_0f92_11eb_b36d_02623cf324c8.slice:cri-containerd:c688c856b21cfb29c1dbf6c14793435e44a1299dfc12add33283239bffed2620
 			if strings.Contains(containerID, "cri-containerd") {
 				strList := strings.Split(containerID, ":")
 				containerID = strList[len(strList)-1]
-				containerID = strings.TrimPrefix(containerID, "cri-containerd-")
 			}
+
+			// remove possible "*-" prefix
+			// e.g. crio-7a92a067289f6197148912be1c15f20f0330c7f3c541473d3b9c4043ca137b42.scope
+			parts := strings.Split(containerID, "-")
```

Comment:
this could panic in `len(parts) == 0`?

Reply:
`strings.Split()` always returns a slice of length ≥ 1, as even the empty string, split on any "split char" would result in `[]string{""}`.

Please [refer to the documentation](https://golang.org/pkg/strings/#Split) where it says:

> If s does not contain sep and sep is not empty, Split returns a slice of length 1 whose only element is s.

Reply:
An it's already tested [here](https://github.com/argoproj/argo/blob/7a4c487ffaf172542f9ba431cf804822d73d0522/workflow/executor/pns/pns_test.go#L15) and [here](https://github.com/argoproj/argo/blob/7a4c487ffaf172542f9ba431cf804822d73d0522/workflow/executor/pns/pns_test.go#L23).

Reply:
My bad - I actually meant line 435 - but I don't think it can panic on closer inspecting